tests/conftest.py

```diff
@@ -21,14 +23,16 @@ def run_uvicorn(uvicorn: Any) -> None:
     uvicorn_run(app, host=uvicorn["host"], port=uvicorn["port"])
 
 
-@pytest.fixture(scope="module")
-def run_server(request: pytest.FixtureRequest) -> Generator[Any, None, None]:
-    uvicorn_settings = request.config.uvicorn  # type: ignore
-
-    process = Process(target=run_uvicorn, args=(uvicorn_settings,))  # type: ignore
-    process.start()
-    yield f"http://{uvicorn_settings['host']}:{uvicorn_settings['port']}"
-    process.terminate()
+@pytest.fixture(scope="session")
+def get_server(request: pytest.FixtureRequest, command_line_args: dict[str, str]) -> Generator[Any, None, None]:
+    if command_line_args["server"]:
+        yield command_line_args["server"]
```

Comment:
its bad practice to yield in 2 places in 1 function

Reply:
Ok, will look you up to discuss how to solve this in a better way.